datumaro/plugins/data_formats/voc/format.py

```diff
@@ -288,20 +329,26 @@ def make_voc_categories(label_map: Optional[LabelMapConfig] = None) -> Categorie
 
     for label, desc in label_map.items():
         label_categories.add(label, attributes=desc[2])
-    for part in OrderedDict((k, None) for k in chain(*(desc[1] for desc in label_map.values()))):
-        label_categories.add(part)
+
+    if task in [VocTask.voc, VocTask.person_layout]:
+        for part in OrderedDict(
+            (k, None) for k in chain(*(desc[1] for desc in label_map.values()))
+        ):
+            label_categories.add(part)
+
     categories[AnnotationType.label] = label_categories
 
-    has_colors = any(v[0] is not None for v in label_map.values())
-    if not has_colors:  # generate new colors
-        colormap = generate_colormap(len(label_map))
-    else:  # only copy defined colors
-        label_id = lambda label: label_categories.find(label)[0]
-        colormap = {
-            label_id(name): desc[0] for name, desc in label_map.items() if desc[0] is not None
-        }
-    mask_categories = MaskCategories(colormap)
-    mask_categories.inverse_colormap  # pylint: disable=pointless-statement
-    categories[AnnotationType.mask] = mask_categories
+    if task in [VocTask.voc, VocTask.segmentation, VocTask.instance_segmentation]:
+        has_colors = any(v[0] is not None for v in label_map.values())
+        if not has_colors:  # generate new colors
+            colormap = generate_colormap(len(label_map))
+        else:  # only copy defined colors
+            label_id = lambda label: label_categories.find(label)[0]
+            colormap = {
+                label_id(name): desc[0] for name, desc in label_map.items() if desc[0] is not None
+            }
+        mask_categories = MaskCategories(colormap)
+        mask_categories.inverse_colormap  # pylint: disable=pointless-statement
+        categories[AnnotationType.mask] = mask_categories
```

Comment:
```suggestion
    if task not in [VocTask.voc, VocTask.segmentation, VocTask.instance_segmentation]:
        return categories

    has_colors = any(v[0] is not None for v in label_map.values())
    if not has_colors:  # generate new colors
        colormap = generate_colormap(len(label_map))
    else:  # only copy defined colors
        label_id = lambda label: label_categories.find(label)[0]
        colormap = {
           label_id(name): desc[0] for name, desc in label_map.items() if desc[0] is not None
        }
    mask_categories = MaskCategories(colormap)
    mask_categories.inverse_colormap  # pylint: disable=pointless-statement
    categories[AnnotationType.mask] = mask_categories
```


Reducing the number of nested lines looks cleaner.